File: app/src/main/java/co/medellinandroid/spotifyartists/di/module/PresentersModule.java
package co.medellinandroid.spotifyartists.di.module;

import co.medellinandroid.spotifyartists.topTenArtists.TopTenArtistsPresenter;
import co.medellinandroid.spotifyartists.topTenSongs.TopTenSongsPresenter;
import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;

@Module public class PresentersModule {

  @Provides @Singleton TopTenArtistsPresenter providesTopTenArtistPresenter() {
    return new TopTenArtistsPresenter();
  }

  @Provides @Singleton TopTenSongsPresenter providesTopTenSongsPresenter() {
    return new TopTenSongsPresenter();
  }
}

File: app/src/main/java/co/medellinandroid/spotifyartists/di/module/PresentersModule.kt
package co.medellinandroid.spotifyartists.di.module

import co.medellinandroid.spotifyartists.topTenArtists.TopTenArtistsPresenter
import co.medellinandroid.spotifyartists.topTenSongs.TopTenSongsPresenter
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class PresentersModule {

  @Provides
  @Singleton internal fun providesTopTenArtistPresenter(): TopTenArtistsPresenter =
      TopTenArtistsPresenter()

  @Provides
  @Singleton internal fun providesTopTenSongsPresenter(): TopTenSongsPresenter =
      TopTenSongsPresenter()
}
